src/actions/instances.ts

```diff
@@ -457,6 +489,90 @@ export const updateInstanceParameterValueNormalized = (index: number, id: Parame
 		}
 	};
 
+export const setInstanceWaitingForMidiMapping = (id: InstanceStateRecord["id"], value: boolean): AppThunk =>
+	(dispatch, getState) => {
+		try {
+			const state = getState();
+			const instance = getInstance(state, id);
+			if (!instance) return;
+
+			dispatch(setInstance(instance.setWatingForMapping(value)));
+
+			try {
+				const message = {
+					address: `${instance.path}/midi/last/report`,
+					args: [
+						{ type: value ? "T" : "F", value: value ? "true" : "false" }
+					]
+				};
+				oscQueryBridge.sendPacket(writePacket(message));
+			} catch (err) {
+				dispatch(showNotification({
+					level: NotificationLevel.error,
+					title: "Error while trying set midi report",
+					message: "Please check the console for further details."
+				}));
+				console.log(err);
+			}
+		} catch (e) {
+			console.log(e);
+		}
+	};
+
+export const updateInstanceMIDIReport = (index: number, value: boolean): AppThunk =>
+	(dispatch, getState) => {
+		try {
+			const state = getState();
+			const instance = getInstanceByIndex(state, index);
+			if (!instance) return;
+			dispatch(setInstance(instance.setWatingForMapping(value)));
+		} catch (e) {
+			console.log(e);
+		}
+	};
+
+export const updateInstanceMIDILastValue = (index: number, value: string): AppThunk =>
+	(dispatch, getState) => {
+		try {
+
+			const state = getState();
+
+			const instance = getInstanceByIndex(state, index);
+			if (!instance?.waitingForMidiMapping) return;
+
+			const midiMeta = JSON.parse(value);
+
+			// find param waiting for mapping
+			const waiting = instance.parameters.filter(p => p.waitingForMidiMapping);
+
+			// stop waiting
+			dispatch(setInstance(instance.clearParametersWaitingForMidiMapping()));
+
+			// update meta
+			waiting.forEach(param => {
+				if (param.waitingForMidiMapping) {
+					let meta: any = {};
+					if (param.meta) {
+						meta = JSON.parse(param.meta);
```

Comment:
This could fail. Seen a few places where we parse the param meta and wondered if we should consolidate this as something like `getParsedMeta` on a param record which has the try catch in the function body, returns a `AnyJson | string` (or other appropriate type - maybe `undefined`) and then the TS typechecks would help with avoiding attempts to set a property on a meta objects that isn't a parsed JSON obj?